src/containers/ConnectedEvent.js

```diff
@@ -0,0 +1,31 @@
+import React, { Component } from "react";
+import Event from "../components/Event";
+import { connect } from "react-redux";
+import queryString from 'query-string'
+import { setEvent } from "../actions/PageActions";
+
+class ConnectedEvent extends Component {
+
+    componentDidUpdate() {
+        const params = queryString.parse(this.props.location.search)
+        const events = this.props.events
+        if (events) {
+            this.props.setEvent(events.find(event => event.id === params.id))
+        }
+    }
+
+    render() {
+        return (
+            <Event event={this.props.event} />
+        );
+    }
+}
+
+const mapStateToProps = state => {
+    return {
+        event: state.pageState.event,
+        events: state.pageState.events
```

Comment:
whats the difference between event and events?

Reply:
Events is a list of event objects
Event is an event object

Maybe it'll be more efficient to make event an event ID instead of an entire object (i think its currently an object i cant remember rn)

Reply:
I think object should be okay for now, can refactor later if noticeable obstacles